src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```diff
@@ -0,0 +1,386 @@
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.esb.connector;
+
+import com.ibm.mq.*;
+import com.ibm.mq.constants.CMQC;
+import com.ibm.mq.constants.MQConstants;
+import com.ibm.mq.headers.MQMD;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.soap.SOAPEnvelope;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.ibm.mq.constants.CMQC.*;
+
+/**
+ * Add messages to queue
+ */
+public class MQPublishQueue extends AbstractConnector {
+
+    MQConnectionBuilder connectionBuilder;
+    MQConfiguration config;
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        try {
+
+            log.info("Message received at ibm-mq connector");
+
+            //retrieve message body
+            SOAPEnvelope soapEnvelope = messageContext.getEnvelope();
+            OMElement getElement = soapEnvelope.getBody().getFirstElement();
+            String queueMessage = "";
+
+            if (getElement != null) {
+                queueMessage = getElement.toString();
+            } else {
+                queueMessage = "";
+            }
+
+            //access queue manager
+            if(connectionBuilder==null || connectionBuilder.getManager()==null) {
+                connectionBuilder = new MQConnectionBuilder(messageContext);
+                config = connectionBuilder.getConfig();
+            }
+
+            MQQueue queue = setQueue(connectionBuilder, config);
+            int messageType = config.getMessageType();
+            MQMessage mqMessage;
+
+            switch (messageType) {
+                case 1:
+                    mqMessage = buildRequestMessageandGetResponse(config, queueMessage);
+                    break;
+                case 4:
+                    mqMessage = buildReportMessage(config, queueMessage);
+                    break;
+                default:
+                    mqMessage = buildDatagramOrReplyMessage(config, queueMessage);
+            }
+
+            if (queue == null || mqMessage == null) {
+                log.error("Error in publishing message");
```

Comment:
Please make this log more informative, what is the exact reason failed publishing the message.